### DISCUSSION TITLE
NFC NDEF handshake?

### DISCUSSION THREAD

**Discussion by supcmd** [decentralized-identity/waci-didcomm#18] [Ideas]
NFC is a much smoother UX ... and is almost identical to the QR flow.  Can this be added as an alternate, using the NDEF protocol?

**Comment by kdenhartog** [comment]
NFC isn't always going to be available because it requires a hardware aspect. Support for NFC should be added, but I wouldn't expect this to be a v0.1 feature. Additionally, it would make sense to move this work over to the DIDComm WG as a way to define how the transport should be used to exchange any DIDComm message.

**Comment by OR13** [comment]
if your flow is URI based, and you embed the URI in an ndef tag, you are kinda done.

I would encourage us try and stick to URIs as much as possible for this reason, especially when considering QR Codes of objects vs QR Codes that encode objects in an URI.

**Comment by OR13** [comment]
Maybe we can add a section to the spec covering NFC recommendations for DIDComm when they are available.

**Comment by kdenhartog** [comment]
Yeah I think this is a perfect example of that the transport extensibility in DIDComm was intentionally designed. My suggestion is to work on something similar to the [DIDComm Bluetooth transport](https://github.com/decentralized-identity/didcomm-bluetooth/blob/main/spec.md) but with NFC instead.

**Comment by brianorwhatever** [comment]
I'm not too familiar with NFC but a problem I see with that flow might be that the invitation must be unique per presentation so the NFC tag or whatever would need to be updated with a new invitation frequently..

**Comment by supcmd** [comment]
NFC could be identical to QR.  Suggestion is to use the NDEF protocol of
NFC which is open and support on most phones, though Apple does not wake up.

**Comment by OR13** [comment]
This appears blocked by DIDComm v2 support for Bluetooth and NFC... See https://github.com/decentralized-identity/didcomm-messaging/issues/222
